src/components/e-date.vue

```diff
@@ -0,0 +1,326 @@
+       *
+       * @returns {Object}
+       */
+      style() {
+        return {
+          '--e-date-format-length': `${this.format.length}ch`,
+        };
+      },
+
+      /**
+       * Calculates the supported input format patterns.
+       *
+       * @returns {Array.<String>}
+       */
+      inputFormats() {
+        const shortFormat = this.format
+          .replace('DD', 'D')
+          .replace('MM', 'M')
+          .replace('YYYY', 'YY');
+
+        return [this.format, shortFormat];
+      },
+
+      /**
+       * Validates, saves and returns the editable string value of the input field.
+       *
+       * @returns {String}
+       */
+      dateString: {
+        set(value) {
+          const date = this.$dayjs(value, this.inputFormats);
+
+          this.invalid = !date.isValid();
+
+          if (this.invalid) {
+            this.textValue = value;
+          } else {
+            this.dateValue = date;
+          }
+        },
+        get() {
+          return this.invalid
+            ? this.textValue
+            : this.dateValue.format(this.format);
+        },
+      },
+    },
+    watch: {
+      /**
+       * Reset the invalid state when the date updates (most likely from the date field).
+       *
+       * @param {Object} newValue - The updated value.
+       * @param {Object} oldValue - The previous value.
+       */
+      dateValue(newValue, oldValue) {
+        if (newValue.toDate().getTime() !== oldValue.toDate().getTime()) {
+          this.invalid = false;
+
+          /**
+           * Sends the current value to the parent.
+           *
+           * @event change
+           *
+           * @property {Date} date - The current date value.
+           */
+          this.$emit('change', this.dateValue.toDate());
+        }
+      },
+
+      /**
+       * Handles updates on the value prop.
+       *
+       * @param {Date} date - The new date value.
+       */
+      value(date) {
+        this.dateValue = this.$dayjs(date);
+      },
+    },
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeDestroy() {},
+    // destroyed() {},
+
+    methods: {
+      /**
+       * Event handler for 'enter' keypress on the input.
+       *
+       * @param {Event} event - The original DOM event.
+       */
+      onEnter(event) {
+        event.preventDefault(); // Prevents auto submit of form.
+        event.target.blur(); // Triggers update.
+      },
+
+      /**
+       * Event handler for the focus event of the input.
+       *
+       * @param {Event} event - The original DOM event.
+       */
+      onFocus(event) {
+        /**
+         * Forwards the focus event to the parent component.
+         *
+         * @event focus
+         *
+         * @property {Event} event - The original DOM event.
+         */
+        this.$emit('focus', event);
```

Comment:
@patric-eberle  Vue warn]: The .native modifier for v-on is only valid on components but it was used on <input>

The warning is indeed correct, but makes difficult to create dynamic component with such events, since the object notation does not permit to specify native modifiers.

https://github.com/vuejs/vue/pull/9884